apps/autoscheduler/FunctionDAG.cpp

```diff
@@ -588,11 +594,9 @@ FunctionDAG::FunctionDAG(const vector<Function> &outputs, const MachineParams &p
     int stage_count = 0;
 
     for (size_t i = order.size(); i > 0; i--) {
-        Function consumer = env[order[i-1]];
-
         Node &node = nodes[order.size() - i];
+        Function consumer = node.func;
```

Comment:
I presume this is intentionally changing from the i'th-from-the-end to always-at-the-end?

Reply:
The loop starting at line 586 ensures that nodes[order.size() - i] and env[order[i-1]] refer to the same function. I can therefore use the function assigned to the node line 588 to reduce the mental load needed to understand this part of the code.

Reply:
ahhh, got it